File: Python/1st.py
def part1(left_list, right_list):
    total_distance = 0
    for i in range(len(left_list)):
        total_distance += abs(left_list[i] - right_list[i])
    return total_distance

def part2(left_list, right_list):
    count_map = {}
    for num in right_list:
        if num in count_map:
            count_map[num] += 1
        else:
            count_map[num] = 1

    # Calculate the similarity score
    similarity_score = 0
    for num in left_list:
        similarity_score += num * count_map.get(num, 0)
    return similarity_score

def main():
    left_list = []
    right_list = []

    with open("/Users/vivekchahal/developer/AdventOfCode/Python/1st.txt", "r") as infile:
        for line in infile:
            left, right = map(int, line.split())
            left_list.append(left)
            right_list.append(right)

    # Sort both lists
    left_list.sort()
    right_list.sort()

    answer1 = part1(left_list, right_list)
    print("Total Distance:", answer1)

    answer2 = part2(left_list, right_list)
    print("Similarity Score:", answer2)

if __name__ == "__main__":
    main()

File: Swift Code/1st.swift
import Foundation

func part1(leftList: [Int], rightList: [Int]) -> Int {
    var totalDistance = 0
    for i in 0..<leftList.count {
        totalDistance += abs(leftList[i] - rightList[i])
    }
    return totalDistance
}

func part2(leftList: [Int], rightList: [Int]) -> Int {
    var countMap: [Int: Int] = [:]
    for num in rightList {
        countMap[num, default: 0] += 1
    }

    // Calculate the similarity score
    var similarityScore = 0
    for num in leftList {
        if let count = countMap[num] {
            similarityScore += num * count
        }
    }
    return similarityScore
}

func main() {
    let filePath = "1st.txt"
    var leftList: [Int] = []
    var rightList: [Int] = []

    // Read pairs from the file and populate the lists
    if let fileContent = try? String(contentsOfFile: "1st.txt", encoding: .utf8) {
        let lines = fileContent.split(separator: "\n")
        for line in lines {
            let numbers = line.split(separator: " ").compactMap { Int($0) }
            if numbers.count == 2 {
                leftList.append(numbers[0])
                rightList.append(numbers[1])
            }
        }
    }

    // Sort both lists
    leftList.sort()
    rightList.sort()

    let answer1 = part1(leftList: leftList, rightList: rightList)
    print("Total Distance: \(answer1)")

    let answer2 = part2(leftList: leftList, rightList: rightList)
    print("Similarity Score: \(answer2)")
}

main()
